### PR TITLE
Improve contrast for error messages

### PR DESCRIPTION
**Pull request by Gymnasiast**
This one was quite easy to pull off and already made quite a difference. Also took the opportunity to name some magic constants.

afbeelding

### CODE REVIEW

**Review by tupaschoal — APPROVED** [review]
Looks great!

**Comment by ZeeMaji** [comment]
I can't help but feel like the color RCT2 uses for this just looks... bad inherently? The black (vanilla) or white (this PR) on bright red just feels out of place and honestly kind of ugly.

I feel like it would look a lot better if you were to expose the error window to theming and color it bordeaux red in a similar manner to the "Demolish ride prompt" window and a few others that are similar, that would make the error window fit in a lot better with the rest of the similar "attention drawing/warning" windows.

Screenshot at 2024-07-23 21-57-47

**Review by AaronVanGeffen — CHANGES_REQUESTED** [review]
Someone pointed out the poor contrast on these error messages on Reddit after the last release. Ultimately, it looks like no issue was filed, so I really appreciate you picking this up.

Having said that, I think white text on a red background increases the contrast too much in the other direction, even with the black outline. The text really stands out too much to me this way; I really do not like it, sorry.

> I feel like it would look a lot better if you were to expose the error window to theming and color it bordeaux red in a similar manner to the "Demolish ride prompt" window and a few others that are similar, that would make the error window fit in a lot better with the rest of the similar "attention drawing/warning" windows and it seems like it would still fix the contrast issue.

This is what I was going to suggest as well. I realise it's going to be a bit more work to adjust the window in this way, but I think it's worth the effort.

**Comment by Gymnasiast** [comment]
> Having said that, I think white text on a red background increases the contrast too much in the other direction, even with the black outline. The text really stands out too much to me this way; I really do not like it, sorry.

I don’t understand this at all - the whole point of it is that the text stands out!

**Comment by AaronVanGeffen** [comment]
Alright. I've pushed changes to make the error window themeable. For the default style, I've copied the style from the demolish ride prompt. Here's what the errors look like:

On grass:
Diamond Heights 2024-07-25 16-48-20

On desert:
Dynamite Dunes 2024-07-25 16-48-36

**Review by AaronVanGeffen — APPROVED** [review]

File: src/openrct2-ui/interface/Theme.cpp [hunk 1]
```diff
@@ -137,6 +137,7 @@ static constexpr WindowThemeDesc WindowThemeDescriptors[] =
     { WindowClass::SavePrompt,                  "WC_SAVE_PROMPT",                    STR_THEMES_WINDOW_SAVE_PROMPT,                    COLOURS_1(translucent(COLOUR_BORDEAUX_RED)                                                                                                             ) },
     { WindowClass::ConstructRide,               "WC_CONSTRUCT_RIDE",                 STR_THEMES_WINDOW_CONSTRUCT_RIDE,                 COLOURS_3(opaque(COLOUR_DARK_BROWN),               opaque(COLOUR_BORDEAUX_RED),             opaque(COLOUR_BORDEAUX_RED)                                ) },
     { WindowClass::DemolishRidePrompt,          "WC_DEMOLISH_RIDE_PROMPT",           STR_THEMES_WINDOW_DEMOLISH_RIDE_PROMPT,           COLOURS_1(translucent(COLOUR_BORDEAUX_RED)                                                                                                             ) },
+    { WindowClass::Error,                       "WC_ERROR",                          STR_THEMES_WINDOW_ERROR,                          COLOURS_1(translucent(COLOUR_BORDEAUX_RED)                                                                                                             ) },
```
Comment (AaronVanGeffen) [inline src/openrct2-ui/interface/Theme.cpp:140]: Bordeaux Red (same as demolish prompt):
Diamond Heights 2024-07-25 16-48-20

Bright Red (looks a bit more error-y?):
Screenshot 2024-07-25 at 18 38 12

Comment (Basssiiie) [reply src/openrct2-ui/interface/Theme.cpp:140]: I would prefer one that meets [accessibility guidelines](https://www.w3.org/WAI/WCAG21/Understanding/contrast-minimum.html):

> The visual presentation of text and images of text has a contrast ratio of at least 4.5:1, [...]

| Scenario | Contrast | 
|--|--|
| Pink text with bordeaux red on grass | [5.4](https://webaim.org/resources/contrastchecker/?fcolor=FFBFBF&bcolor=8F2727) |
| Pink text with bright red on grass | [4.76:1](https://webaim.org/resources/contrastchecker/?fcolor=FFDBD7&bcolor=C70200) |
| Pink text with bordeaux red on sand | [3.24:1](https://webaim.org/resources/contrastchecker/?fcolor=FFBFBF&bcolor=B34F4F) |
| Pick text with bright red on sand | ? | 

Some more tests would be nice but I'm not available in the coming days to test it ingame myself unfortunately. A test with snow would be interesting too. 😊

**Comment by Gymnasiast** [comment]
Thanks for make the changes requested. I’d like to take a look myself before merging, but I can’t self-request a review on my own PR obviously :')

File: src/openrct2-ui/interface/Theme.cpp (hunk 1 above)
Comment (Gymnasiast) [reply src/openrct2-ui/interface/Theme.cpp:140]: Not sure if that’s really necessary, as people can now change it themselves. If people have trouble reading, I expect them to disable the transparent background to begin with.

**Comment by Gymnasiast** [comment]
Tested it. One thing I did notice is that error message can be put partly offscreen, even though there is code that supposedly prevents that:

afbeelding

**Comment by AaronVanGeffen** [comment]
Ah, I think that might be due to the window being positioned before the initial draw? I can take a look tonight.

**Comment by janisozaur** [comment]
> Someone pointed out the poor contrast on these error messages on Reddit after the last release. Ultimately, it looks like no issue was filed, so I really appreciate you picking this up.

#22250

**Pull request merged**